tests/worker/test_scheduler.py

```diff
@@ -12,52 +12,99 @@
 
 
 @pytest.fixture
-def scheduler() -> Iterator[Scheduler]:
+async def scheduler() -> Iterator[Scheduler]:
     _scheduler = Scheduler()
     yield _scheduler
-    _scheduler.close()
+    await _scheduler.close()
 
 
 @pytest.mark.asyncio
 async def test_scheduler(scheduler: Scheduler) -> None:
     queue1 = MagicMock(spec=Queue)
-    queue1.get.return_value = sentinel.queue1
+    queue1.iterator.return_value = alib.cycle([sentinel.queue1])
     queue2 = MagicMock(spec=Queue)
-    queue2.get.return_value = sentinel.queue2
+    queue2.iterator.return_value = alib.cycle([sentinel.queue2])
 
     scheduler.add(queue1)
     scheduler.add(queue2)
 
     messages: Counter[Message] = Counter()
-    async for item, _ in alib.zip(scheduler.iter(), range(10)):
+    scheduler_iter = scheduler.iter()
+
+    async for item, _ in alib.zip(alib.borrow(scheduler_iter), range(10)):
```

Comment:
Seems to be something from `asyncstdlib` with no equivalent in the stdlib. In any case, this is useful here to avoid `zip` closing the original generator. (I might have spent way more time than I wish debugging the initial code with repeated call to `scheduler.iter()`)